include/sustainml/Engine.h

```diff
@@ -153,20 +148,19 @@ public slots:
             const types::TaskId& task_id);
 
     QString get_raw_output(
-            const sustainml::NodeID& id,
-            void* data);
+            const QJsonObject& json);
 
     QString update_node_status(
             const sustainml::NodeID& id,
             const types::NodeStatus& status);
 
     size_t split_string(
             const std::string& string,
-            std::vector<std::string>& string_set,
+            QJsonArray& string_array,
             char delimeter);
 
-
-    sustainml::orchestrator::OrchestratorNode* orchestrator;
+    QNetworkAccessManager* user_input_request_;
+    std::array<QNetworkAccessManager*, static_cast<size_t>(sustainml::NodeID::MAX)> node_responses_;
```

Comment:
Same comment, I think that these pointers are not being deleted when the application exits